packages/graphql/src/schema/make-augmented-schema.ts

```diff
@@ -528,6 +624,56 @@ function makeAugmentedSchema(
             },
         });
 
+        composer.createObjectTC({
+            name: `${node.name}AggregateSelection`,
+            fields: {
+                count: composeInt,
+                ...[...node.primitiveFields, ...node.temporalFields].reduce((res, field) => {
+                    if (field.typeMeta.array) {
+                        return res;
+                    }
+
+                    if (field.typeMeta.name === "ID") {
+                        res[field.fieldName] = idNumericalAggregationSelection.NonNull;
+                    }
+
+                    if (field.typeMeta.name === "String") {
+                        res[field.fieldName] = stringNumericalAggregationSelection.NonNull;
+                    }
+
+                    if (field.typeMeta.name === "Float") {
+                        res[field.fieldName] = floatNumericalAggregationSelection.NonNull;
+                    }
+
+                    if (field.typeMeta.name === "Int") {
+                        res[field.fieldName] = intNumericalAggregationSelection.NonNull;
+                    }
+
+                    if (field.typeMeta.name === "BigInt") {
+                        res[field.fieldName] = bigIntNumericalAggregationSelection.NonNull;
+                    }
+
+                    if (field.typeMeta.name === "DateTime") {
+                        res[field.fieldName] = dateTimeAggregationSelection.NonNull;
+                    }
+
+                    if (field.typeMeta.name === "LocalDateTime") {
+                        res[field.fieldName] = localDateTimeAggregationSelection.NonNull;
+                    }
+
+                    if (field.typeMeta.name === "LocalTime") {
+                        res[field.fieldName] = localTimeAggregationSelection.NonNull;
+                    }
+
+                    if (field.typeMeta.name === "Time") {
+                        res[field.fieldName] = timeAggregationSelection.NonNull;
+                    }
```

Comment:
This is a bit clunky - is it worthwhile having an object like the following above:

```ts
const aggregateSelections = {
  ID: idNumericalAggregationSelection.NonNull,
  String: stringNumericalAggregationSelection.NonNull,
  ...
};
```

And then referring to it using:

```ts
res[field.fieldName] = aggregateSelections[field.typeMeta.name];
```

Just a suggestion, a bit tidier perhaps?

Reply:
Good observation, I have made a change now to make this less repetitive.